File: problems/3438-find-valid-pair-of-adjacent-digits-in-string/solution.java
class Solution {
    public String findValidPair(String s) {
        int n = s.length();
        for (int i = 0; i < n - 1; i++) {
            char first = s.charAt(i);
            char second = s.charAt(i + 1);

            if (first != second) {
                int firstDigit = first - '0';
                int secondDigit = second - '0';

                int firstCount = 0;
                int secondCount = 0;

                for (int j = 0; j < n; j++) {
                    if (s.charAt(j) == first) {
                        firstCount++;
                    }
                    if (s.charAt(j) == second) {
                        secondCount++;
                    }
                }

                if (firstCount == firstDigit && secondCount == secondDigit) {
                    return "" + first + second;
                }
            }
        }
        return "";
    }
}

File: problems/3438-find-valid-pair-of-adjacent-digits-in-string/solution.cpp
class Solution {
public:
    string findValidPair(string s) {
        int counts[10] = {0};
        for (char c : s) {
            counts[c - '0']++;
        }

        for (int i = 0; i < s.length() - 1; ++i) {
            if (s[i] != s[i+1]) {
                int firstDigit = s[i] - '0';
                int secondDigit = s[i+1] - '0';

                if (counts[firstDigit] == firstDigit && counts[secondDigit] == secondDigit) {
                    return string(1, s[i]) + string(1, s[i+1]);
                }
            }
        }

        return "";
    }
};